### PR TITLE
clear mkldnn dependency remained after transfer_layout migrated

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

Others
### PR changes

OPs
### Describe

When migrating transfer_layout #45261  (commit: 985f2a4a019e438f2ac4edfc10683e781f2f1ea5), some mkldnn functions were copied to `phi` directory, while the code in `fluid` directory was not deleted. This pr is intended to clear these code and their dependencies.

### CODE REVIEW

**Review by YuanRisheng — DISMISSED** [review]

File: paddle/fluid/operators/mkldnn/split_mkldnn_op.cc [hunk 1]
```diff
@@ -95,18 +96,14 @@ class SplitMKLDNNKernel : public framework::OpKernel<T> {
...
     auto& astream = platform::MKLDNNDeviceContext::tls().get_stream();
 
     std::vector<int64_t> offset(x_vec_dims.size(), 0);
 
-    platform::ReorderMKLDNNHandler reorder_handler(
-        x_vec_dims,
-        framework::TransToProtoVarType(x->dtype()),
-        x_type,
-        onednn_engine);
+    phi::funcs::ReorderMKLDNNHandler reorder_handler(
```
Comment (chenwhql) [inline paddle/fluid/operators/mkldnn/split_mkldnn_op.cc]: maybe ReorderMKLDNNHandler put in `phi/backends/onednn` is better? It is a onednn-library related class, so it is put in `platform` dir originally

**Review by zyfncg — DISMISSED** [review]
LGTM

**Review by chenwhql — COMMENTED** [review]
maybe mkldnn_helper.h and mkldnn_reuse.h are better move into phi/backends/onednn

**Comment by jczaja** [comment]
@piotrekobi Please help with review

**Review by piotrekobi — DISMISSED** [review]
This should make migrating some oneDNN kernels to PHI easier. Thanks for your effort. LGTM!

**Comment by kangguangli** [comment]
@piotrekobi Thanks for your review. I noticed that you're moving these files #45626 too, so please  give attention to this pr #45659, we(@chenwhql) decided to move the onednn files from phi/kernels/funcs to phi/backends. 
As for this pr, i'll continue after #45659 and yours  #45626 ends to avoid too many conflicts.

File: paddle/fluid/operators/mkldnn/split_mkldnn_op.cc (hunk 1 above)
Comment (kangguangli) [reply paddle/fluid/operators/mkldnn/split_mkldnn_op.cc]: done in #45659

**Review by piotrekobi — APPROVED** [review]
LGTM. Thanks again for making the migration of oneDNN kernels to PHI easier for us 😃

**Comment by kangguangli** [comment]
Outdated

**Pull request closed without merging**